servers/src/grin/sync/header_sync.rs

```diff
@@ -66,6 +64,15 @@ impl HeaderSync {
 					header_head.height,
 				);
 
+				// Reset sync_head to header_head on transition to HeaderSync,
+				// but ONLY on initial transition to HeaderSync state.
+				{
+					let store = self.chain.store();
+					let batch = store.batch().unwrap();
+					batch.reset_sync_head().unwrap();
+					batch.commit().unwrap();
+				}
+
```

Comment:
Need move that debug log down (after this reset), because it says `reset to: {} at {}`:

```
				debug!(
					"sync: initial transition to HeaderSync. sync_head: {} at {}, reset to: {} at {}",
					sync_head.hash(),
					sync_head.height,
					header_head.hash(),
					header_head.height,
				);
```

Reply:
Doesn't make sense to log this _after_ updating the `sync_head` as we log `header_head` and `sync_head` (and they will be identical after the reset).